pkg/reconciler/common/env_override.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/operator/pkg/apis/operator/base"
+)
+
+func mergeEnv(src, tgt *[]v1.EnvVar) {
+	if len(*tgt) > 0 {
+		for _, srcV := range *src {
+			for i, tgtV := range *tgt {
+				if srcV.Name == tgtV.Name {
+					(*tgt)[i] = srcV
+				}
+			}
+		}
+	} else {
+		*tgt = *src
+	}
+}
+
+func findEnvOverride(resources []base.EnvRequirementsOverride, name string) *base.EnvRequirementsOverride {
+	for _, override := range resources {
+		if override.Container == name {
+			return &override
+		}
+	}
+	return nil
+}
+
+func getEnv(containers []v1.Container, container string) []v1.EnvVar {
```

Comment:
This function is used by unit test only (`deployments_override_test.go`). It should be moved to `deployments_override_test.go`?